sklearn/utils/estimator_checks.py

```diff
@@ -196,6 +212,73 @@ def _boston_subset(n_samples=200):
     return BOSTON
 
 
+def _readonly_boston_subset(n_samples=200):
+    """Utility function used to return a r-o memmap, without recreating a new memory map at each call"""
+    _init_temp_memory()
+    f = _TEMP_MEMORY.cache(_boston_subset)
+    return f(n_samples=n_samples)
+
+
+def _boston_subset_with_mode(readonly=False):
+    """Factorisation function used in checks"""
+    if readonly:
+        return _readonly_boston_subset()
+    else:
+        return _boston_subset()
+
+
+def _make_blobs(*args, **kwargs):
+    """Utility function used to ensure that we have only positive value for X"""
+    positive = kwargs.pop('positive', False)
+    scale = kwargs.pop('scale', False)
+    X, y = make_blobs(*args, **kwargs)
+    if scale:
+        X = StandardScaler().fit_transform(X)
+    if positive:
+        X -= X.min()
+    return X, y
+
+
+def _readonly_make_blobs(*args, **kwargs):
+    """Utility function used to return a r-o memmap, without recreating a new memory map at each call"""
+    _init_temp_memory()
+    f = _TEMP_MEMORY.cache(_make_blobs)
+    return f(*args, **kwargs)
+
+
+def _make_blobs_with_mode(*args, **kwargs):
+    """Factorisation function used in checks"""
+    readonly = kwargs.pop('readonly', None)
+    if readonly:
+        return _readonly_make_blobs(*args, **kwargs)
+    else:
+        return _make_blobs(*args, **kwargs)
+
+
+def _init_temp_memory():
```

Comment:
I would pass `mmap_mode='r'` as kwarg to this function to make it explicit that this is a readonly memmap cache and not just an implementation detail.